internal/common/function/function.go

```diff
@@ -77,66 +77,125 @@ func ValidateBackupSpec(nonAdminBackup *nacv1alpha1.NonAdminBackup) error {
 	return nil
 }
 
-// GenerateVeleroBackupName generates a Velero backup name based on the provided namespace and NonAdminBackup name.
-// It calculates a hash of the NonAdminBackup name and combines it with the namespace and a prefix to create the Velero backup name.
-// If the resulting name exceeds the maximum Kubernetes name length, it truncates the namespace to fit within the limit.
-func GenerateVeleroBackupName(namespace, nabName string) string {
-	// Calculate a hash of the name
-	hasher := sha256.New()
-	_, err := hasher.Write([]byte(nabName))
-	if err != nil {
-		return ""
+// GenerateNacObjectNameWithUUID generates a unique name based on the provided namespace and object origin name.
+// It includes a UUID suffix. If the name exceeds the maximum length, it truncates nacName first, then namespace.
+func GenerateNacObjectNameWithUUID(namespace, nacName string) string {
```

Comment:
this name does not make sense to me. This does not generate NAB/NAR name, it generates Velero backup/restore name, right? `GenerateVeleroObjectName` makes more sense?